Tests fail when Torch not installed

I assume this broke with #345. @gheinrich can you confirm?

I wonder if we can setup TravisCI to run the tests with AND without Torch ...
```
1) ERROR: test suite for <class 'digits.model.images.generic.test_views.TestTorchCreated'>

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/suite.py line 209 in run
      self.setUp()
    venv/lib/python2.7/site-packages/nose/suite.py line 292 in setUp
      self.setupContext(ancestor)
    venv/lib/python2.7/site-packages/nose/suite.py line 315 in setupContext
      try_run(context, names)
    venv/lib/python2.7/site-packages/nose/util.py line 471 in try_run
      return func()
    digits/model/images/generic/test_views.py line 200 in setUpClass
      cls.model_id = cls.create_model(json=True)
    digits/model/images/generic/test_views.py line 173 in create_model
      raise RuntimeError('Model creation failed with %s' % rv.status_code)
   RuntimeError: Model creation failed with 500
   -------------------- >> begin captured logging << --------------------
   digits.webapp: INFO: Analyze DB (Training Images) task started.
   digits.webapp: INFO: Analyze DB (Training Labels) task started.
   digits.webapp: INFO: Analyze DB (Validation Images) task started.
   digits.webapp: INFO: Analyze DB (Validation Labels) task started.
   digits.webapp: WARNING: Analyze DB (Training Labels) unrecognized output: WARNING: log_file config option not found - no log file is being saved
   digits.webapp: DEBUG: Images are 2x1x1
   digits.webapp: WARNING: Analyze DB (Validation Images) unrecognized output: WARNING: log_file config option not found - no log file is being saved
   digits.webapp: DEBUG: Images are 10x10x1
   digits.webapp: WARNING: Analyze DB (Training Images) unrecognized output: WARNING: log_file config option not found - no log file is being saved
   digits.webapp: DEBUG: Images are 10x10x1
   digits.webapp: INFO: Analyze DB (Training Labels) task completed.
   digits.webapp: INFO: Analyze DB (Validation Images) task completed.
   digits.webapp: INFO: Analyze DB (Training Images) task completed.
   digits.webapp: WARNING: Analyze DB (Validation Labels) unrecognized output: WARNING: log_file config option not found - no log file is being saved
   digits.webapp: DEBUG: Images are 2x1x1
   digits.webapp: INFO: Analyze DB (Validation Labels) task completed.
   digits: INFO: Job complete.
   --------------------- >> end captured logging << ---------------------

-----------------------------------------------------------------------------
2) ERROR: digits.model.images.generic.test_views.TestTorchCreation.test_create_abort_delete

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/case.py line 197 in runTest
      self.test(*self.arg)
    digits/model/images/generic/test_views.py line 250 in test_create_abort_delete
      job_id = self.create_model()
    digits/model/images/generic/test_views.py line 184 in create_model
      raise RuntimeError('Failed to create model')
   RuntimeError: Failed to create model

   -------------------- >> begin captured stdout << ---------------------
   Status code: 500
   
   --------------------- >> end captured stdout << ----------------------
   -------------------- >> begin captured logging << --------------------
   digits.webapp: INFO: Analyze DB (Training Images) task started.
   digits.webapp: INFO: Analyze DB (Training Labels) task started.
   digits.webapp: INFO: Analyze DB (Validation Images) task started.
   digits.webapp: INFO: Analyze DB (Validation Labels) task started.
   digits.webapp: WARNING: Analyze DB (Training Labels) unrecognized output: WARNING: log_file config option not found - no log file is being saved
   digits.webapp: DEBUG: Images are 2x1x1
   digits.webapp: WARNING: Analyze DB (Validation Images) unrecognized output: WARNING: log_file config option not found - no log file is being saved
   digits.webapp: DEBUG: Images are 10x10x1
   digits.webapp: WARNING: Analyze DB (Training Images) unrecognized output: WARNING: log_file config option not found - no log file is being saved
   digits.webapp: DEBUG: Images are 10x10x1
   digits.webapp: INFO: Analyze DB (Training Labels) task completed.
   digits.webapp: INFO: Analyze DB (Validation Images) task completed.
   digits.webapp: INFO: Analyze DB (Training Images) task completed.
   digits.webapp: WARNING: Analyze DB (Validation Labels) unrecognized output: WARNING: log_file config option not found - no log file is being saved
   digits.webapp: DEBUG: Images are 2x1x1
   digits.webapp: INFO: Analyze DB (Validation Labels) task completed.
   digits: INFO: Job complete.
   digits.webapp: INFO: Analyze DB (Training Images) task started.
   digits.webapp: INFO: Analyze DB (Training Labels) task started.
   digits.webapp: INFO: Analyze DB (Validation Images) task started.
   digits.webapp: INFO: Analyze DB (Validation Labels) task started.
   digits.webapp: WARNING: Analyze DB (Training Images) unrecognized output: WARNING: log_file config option not found - no log file is being saved
   digits.webapp: DEBUG: Images are 10x10x1
   digits.webapp: INFO: Analyze DB (Training Images) task completed.
   digits.webapp: WARNING: Analyze DB (Training Labels) unrecognized output: WARNING: log_file config option not found - no log file is being saved
   digits.webapp: DEBUG: Images are 2x1x1
   digits.webapp: WARNING: Analyze DB (Validation Labels) unrecognized output: WARNING: log_file config option not found - no log file is being saved
   digits.webapp: DEBUG: Images are 2x1x1
   digits.webapp: WARNING: Analyze DB (Validation Images) unrecognized output: WARNING: log_file config option not found - no log file is being saved
   digits.webapp: DEBUG: Images are 10x10x1
   digits.webapp: INFO: Analyze DB (Training Labels) task completed.
   digits.webapp: INFO: Analyze DB (Validation Images) task completed.
   digits.webapp: INFO: Analyze DB (Validation Labels) task completed.
   digits: INFO: Job complete.
   --------------------- >> end captured logging << ---------------------

-----------------------------------------------------------------------------
3) ERROR: digits.model.images.generic.test_views.TestTorchCreation.test_create_delete

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/case.py line 197 in runTest
      self.test(*self.arg)
    digits/model/images/generic/test_views.py line 239 in test_create_delete
      job_id = self.create_model()
    digits/model/images/generic/test_views.py line 184 in create_model
      raise RuntimeError('Failed to create model')
   RuntimeError: Failed to create model

   -------------------- >> begin captured stdout << ---------------------
   Status code: 500
   
   --------------------- >> end captured stdout << ----------------------

-----------------------------------------------------------------------------
4) ERROR: digits.model.images.generic.test_views.TestTorchCreation.test_create_json

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/case.py line 197 in runTest
      self.test(*self.arg)
    digits/model/images/generic/test_views.py line 235 in test_create_json
      job_id = self.create_model(json=True)
    digits/model/images/generic/test_views.py line 173 in create_model
      raise RuntimeError('Model creation failed with %s' % rv.status_code)
   RuntimeError: Model creation failed with 500

   -------------------- >> begin captured stdout << ---------------------
   {u'error': {u'message': u"'NoneType' object has no attribute 'validate_network'", u'type': u'AttributeError', u'trace': [u'Traceback (most recent call last):', u'  File "/usr/lib/python2.7/dist-packages/flask/app.py", line 1475, in full_dispatch_request', u'    rv = self.dispatch_request()', u'  File "/usr/lib/python2.7/dist-packages/flask/app.py", line 1461, in dispatch_request', u'    return self.view_functions[rule.endpoint](**req.view_args)', u'  File "/home/lyeager/digits/digits/model/images/generic/views.py", line 62, in generic_image_model_create', u'    if not form.validate_on_submit():', u'  File "/usr/lib/python2.7/dist-packages/flask_wtf/form.py", line 166, in validate_on_submit', u'    return self.is_submitted() and self.validate()', u'  File "/usr/lib/python2.7/dist-packages/wtforms/form.py", line 310, in validate', u'    return super(Form, self).validate(extra)', u'  File "/usr/lib/python2.7/dist-packages/wtforms/form.py", line 152, in validate', u'    if not field.validate(self, extra):', u'  File "/usr/lib/python2.7/dist-packages/wtforms/fields/core.py", line 200, in validate', u'    stop_validation = self._run_validation_chain(form, chain)', u'  File "/usr/lib/python2.7/dist-packages/wtforms/fields/core.py", line 220, in _run_validation_chain', u'    validator(form, self)', u'  File "/home/lyeager/digits/digits/model/forms.py", line 32, in validate_NetParameter', u'    fw.validate_network(field.data)', u"AttributeError: 'NoneType' object has no attribute 'validate_network'", u'']}}
   
   --------------------- >> end captured stdout << ----------------------

-----------------------------------------------------------------------------
5) ERROR: digits.model.images.generic.test_views.TestTorchCreation.test_create_wait_delete

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/case.py line 197 in runTest
      self.test(*self.arg)
    digits/model/images/generic/test_views.py line 244 in test_create_wait_delete
      job_id = self.create_model()
    digits/model/images/generic/test_views.py line 184 in create_model
      raise RuntimeError('Failed to create model')
   RuntimeError: Failed to create model

   -------------------- >> begin captured stdout << ---------------------
   Status code: 500
   
   --------------------- >> end captured stdout << ----------------------

-----------------------------------------------------------------------------
6) ERROR: digits.model.images.generic.test_views.TestTorchCreation.test_retrain

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/case.py line 197 in runTest
      self.test(*self.arg)
    digits/model/images/generic/test_views.py line 309 in test_retrain
      job1_id = self.create_model()
    digits/model/images/generic/test_views.py line 184 in create_model
      raise RuntimeError('Failed to create model')
   RuntimeError: Failed to create model

   -------------------- >> begin captured stdout << ---------------------
   Status code: 500
   
   --------------------- >> end captured stdout << ----------------------

-----------------------------------------------------------------------------
7) ERROR: digits.model.images.generic.test_views.TestTorchCreation.test_retrain_twice

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/case.py line 197 in runTest
      self.test(*self.arg)
    digits/model/images/generic/test_views.py line 327 in test_retrain_twice
      job1_id = self.create_model()
    digits/model/images/generic/test_views.py line 184 in create_model
      raise RuntimeError('Failed to create model')
   RuntimeError: Failed to create model

   -------------------- >> begin captured stdout << ---------------------
   Status code: 500
   
   --------------------- >> end captured stdout << ----------------------

-----------------------------------------------------------------------------
8) ERROR: digits.model.images.generic.test_views.TestTorchCreation.test_select_gpus(('0',),)

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/case.py line 197 in runTest
      self.test(*self.arg)
    digits/model/images/generic/test_views.py line 305 in check_select_gpus
      job_id = self.create_model(select_gpus_list=','.join(gpu_list), batch_size=len(gpu_list))
    digits/model/images/generic/test_views.py line 184 in create_model
      raise RuntimeError('Failed to create model')
   RuntimeError: Failed to create model

   -------------------- >> begin captured stdout << ---------------------
   Status code: 500
   
   --------------------- >> end captured stdout << ----------------------

-----------------------------------------------------------------------------
9) ERROR: digits.model.images.generic.test_views.TestTorchCreation.test_select_gpus(('1',),)

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/case.py line 197 in runTest
      self.test(*self.arg)
    digits/model/images/generic/test_views.py line 305 in check_select_gpus
      job_id = self.create_model(select_gpus_list=','.join(gpu_list), batch_size=len(gpu_list))
    digits/model/images/generic/test_views.py line 184 in create_model
      raise RuntimeError('Failed to create model')
   RuntimeError: Failed to create model

   -------------------- >> begin captured stdout << ---------------------
   Status code: 500
   
   --------------------- >> end captured stdout << ----------------------

-----------------------------------------------------------------------------
10) ERROR: digits.model.images.generic.test_views.TestTorchCreation.test_select_gpus(('2',),)

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/case.py line 197 in runTest
      self.test(*self.arg)
    digits/model/images/generic/test_views.py line 305 in check_select_gpus
      job_id = self.create_model(select_gpus_list=','.join(gpu_list), batch_size=len(gpu_list))
    digits/model/images/generic/test_views.py line 184 in create_model
      raise RuntimeError('Failed to create model')
   RuntimeError: Failed to create model

   -------------------- >> begin captured stdout << ---------------------
   Status code: 500
   
   --------------------- >> end captured stdout << ----------------------

-----------------------------------------------------------------------------
11) ERROR: digits.model.images.generic.test_views.TestTorchCreation.test_select_gpus(('0', '1'),)

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/case.py line 197 in runTest
      self.test(*self.arg)
    digits/model/images/generic/test_views.py line 305 in check_select_gpus
      job_id = self.create_model(select_gpus_list=','.join(gpu_list), batch_size=len(gpu_list))
    digits/model/images/generic/test_views.py line 184 in create_model
      raise RuntimeError('Failed to create model')
   RuntimeError: Failed to create model

   -------------------- >> begin captured stdout << ---------------------
   Status code: 500
   
   --------------------- >> end captured stdout << ----------------------

-----------------------------------------------------------------------------
12) ERROR: digits.model.images.generic.test_views.TestTorchCreation.test_select_gpus(('0', '2'),)

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/case.py line 197 in runTest
      self.test(*self.arg)
    digits/model/images/generic/test_views.py line 305 in check_select_gpus
      job_id = self.create_model(select_gpus_list=','.join(gpu_list), batch_size=len(gpu_list))
    digits/model/images/generic/test_views.py line 184 in create_model
      raise RuntimeError('Failed to create model')
   RuntimeError: Failed to create model

   -------------------- >> begin captured stdout << ---------------------
   Status code: 500
   
   --------------------- >> end captured stdout << ----------------------

-----------------------------------------------------------------------------
13) ERROR: digits.model.images.generic.test_views.TestTorchCreation.test_select_gpus(('1', '2'),)

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/case.py line 197 in runTest
      self.test(*self.arg)
    digits/model/images/generic/test_views.py line 305 in check_select_gpus
      job_id = self.create_model(select_gpus_list=','.join(gpu_list), batch_size=len(gpu_list))
    digits/model/images/generic/test_views.py line 184 in create_model
      raise RuntimeError('Failed to create model')
   RuntimeError: Failed to create model

   -------------------- >> begin captured stdout << ---------------------
   Status code: 500
   
   --------------------- >> end captured stdout << ----------------------

-----------------------------------------------------------------------------
14) ERROR: digits.model.images.generic.test_views.TestTorchCreation.test_select_gpus(('0', '1', '2'),)

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/case.py line 197 in runTest
      self.test(*self.arg)
    digits/model/images/generic/test_views.py line 305 in check_select_gpus
      job_id = self.create_model(select_gpus_list=','.join(gpu_list), batch_size=len(gpu_list))
    digits/model/images/generic/test_views.py line 184 in create_model
      raise RuntimeError('Failed to create model')
   RuntimeError: Failed to create model

   -------------------- >> begin captured stdout << ---------------------
   Status code: 500
   
   --------------------- >> end captured stdout << ----------------------

-----------------------------------------------------------------------------
15) ERROR: digits.model.images.generic.test_views.TestTorchCreation.test_snapshot_interval_0_5

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/case.py line 197 in runTest
      self.test(*self.arg)
    digits/model/images/generic/test_views.py line 264 in test_snapshot_interval_0_5
      job_id = self.create_model(train_epochs=4, snapshot_interval=2)
    digits/model/images/generic/test_views.py line 184 in create_model
      raise RuntimeError('Failed to create model')
   RuntimeError: Failed to create model

   -------------------- >> begin captured stdout << ---------------------
   Status code: 500
   
   --------------------- >> end captured stdout << ----------------------

-----------------------------------------------------------------------------
16) ERROR: digits.model.images.generic.test_views.TestTorchCreation.test_snapshot_interval_2

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/case.py line 197 in runTest
      self.test(*self.arg)
    digits/model/images/generic/test_views.py line 256 in test_snapshot_interval_2
      job_id = self.create_model(snapshot_interval=0.5)
    digits/model/images/generic/test_views.py line 184 in create_model
      raise RuntimeError('Failed to create model')
   RuntimeError: Failed to create model

   -------------------- >> begin captured stdout << ---------------------
   Status code: 500
   
   --------------------- >> end captured stdout << ----------------------

-----------------------------------------------------------------------------
17) ERROR: digits.model.images.generic.test_views.TestTorchDatasetModelInteractions.test_create_model_running_dataset

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/case.py line 197 in runTest
      self.test(*self.arg)
    digits/model/images/generic/test_views.py line 489 in test_create_model_running_dataset
      model_id = self.create_model(dataset=dataset_id)
    digits/model/images/generic/test_views.py line 184 in create_model
      raise RuntimeError('Failed to create model')
   RuntimeError: Failed to create model

   -------------------- >> begin captured stdout << ---------------------
   Status code: 500
   
   --------------------- >> end captured stdout << ----------------------

-----------------------------------------------------------------------------
18) ERROR: digits.model.images.generic.test_views.TestTorchDatasetModelInteractions.test_delete_dataset_dependent_model

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/case.py line 197 in runTest
      self.test(*self.arg)
    digits/model/images/generic/test_views.py line 520 in test_delete_dataset_dependent_model
      model_id = self.create_model(dataset=dataset_id)
    digits/model/images/generic/test_views.py line 184 in create_model
      raise RuntimeError('Failed to create model')
   RuntimeError: Failed to create model

   -------------------- >> begin captured stdout << ---------------------
   Status code: 500
   
   --------------------- >> end captured stdout << ----------------------

-----------------------------------------------------------------------------
19) ERROR: digits.model.images.generic.test_views.TestTorchDatasetModelInteractions.test_delete_running_dataset_dependent_model

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/case.py line 197 in runTest
      self.test(*self.arg)
    digits/model/images/generic/test_views.py line 528 in test_delete_running_dataset_dependent_model
      model_id = self.create_model(dataset=dataset_id)
    digits/model/images/generic/test_views.py line 184 in create_model
      raise RuntimeError('Failed to create model')
   RuntimeError: Failed to create model

   -------------------- >> begin captured stdout << ---------------------
   Status code: 500
   
   --------------------- >> end captured stdout << ----------------------

-----------------------------------------------------------------------------
20) FAIL: digits.model.images.generic.test_views.TestTorchViews.test_visualize_network

   Traceback (most recent call last):
    venv/lib/python2.7/site-packages/nose/case.py line 197 in runTest
      self.test(*self.arg)
    digits/model/images/generic/test_views.py line 225 in test_visualize_network
      raise AssertionError('POST failed with %s\n\n%s' % (rv.status_code, body))
   AssertionError: POST failed with 500
   
   <body>
   <h2>AttributeError</h2>
   <p class="text-danger">'NoneType' object has no attribute 'get_network_visualization'</p>
   <pre class="text-muted">
   Traceback (most recent call last):
     File "/usr/lib/python2.7/dist-packages/flask/app.py", line 1475, in full_dispatch_request
       rv = self.dispatch_request()
     File "/usr/lib/python2.7/dist-packages/flask/app.py", line 1461, in dispatch_request
       return self.view_functions[rule.endpoint](**req.view_args)
     File "/home/lyeager/digits/digits/model/views.py", line 164, in models_visualize_network
       ret = fw.get_network_visualization(flask.request.form['custom_network'])
   AttributeError: 'NoneType' object has no attribute 'get_network_visualization'
   
   </pre>
   </body>


-----------------------------------------------------------------------------
684 tests run in 276.1 seconds. 
1 FAILED, 19 errors, 9 skipped (655 tests passed)
```

--- a/digits/model/images/generic/test_views.py
+++ b/digits/model/images/generic/test_views.py
@@ -80,6 +80,11 @@ return function(params)
     }
 end
 """
+    @classmethod
+    def setUpClass(cls):
+        super(BaseViewsTest, cls).setUpClass()
+        if cls.FRAMEWORK=='torch' and not config_value('torch_root'):
+            raise unittest.SkipTest('Torch not found')
 
     @classmethod
     def model_exists(cls, job_id):

